MVVM-C/ViewModel/ArticleViewModel.swift

```diff
@@ -0,0 +1,60 @@
+//
+//  ArticleViewModel.swift
+//  MVVM-C
+//
+//  Created by Usemobile on 09/06/20.
+//  Copyright © 2020 Jakub Homik. All rights reserved.
+//
+
+import Foundation
+
+// ViewModel: Its responsible for holding the model and formatting it to be displayed to UI component, holding state
+
+/*
+ Model                     <---> ViewModel <---> UI
+ (Article, ArticleManager) <---> ViewModel <---> (GotCell)
+ */
+
+final class ArticleViewModel {
+    
+    private let model: Article
+    
+    typealias FavoriteObserver = ((Bool) -> Void)
+    var onFavorite = [NSObject: FavoriteObserver]()
```

Comment:
also this propoerty onFavorite, and FavoriteObserver. I know that the second one is a closure but I can't figured it out how this work. Its really advanced way of doing this

Reply:
First I've created a typealias to give a name to my observer, so, here I have an observer called `FavoriteObserver` which receives a Boolean and returns Void.
Than I've created a property that is a dictionary where the Key must be a NSObject and the value is a `FavoriteObserver`, so that I can store more than one observer and trigger all of them when my `isFavorite` boolean changes.

Reply:
What do you mean by saying Observer? What is this observer here ??

Reply:
Observer is a code block that will trigger an action whenever it changes.